### ISSUE TITLE
ASPI causes a crash on some systems. (Bugzilla #2)

### ISSUE THREAD

**Issue by GerbilSoft** [GerbilSoft/gens-gs-ii#2]
status RESOLVED severity _critical_ in component _gens_ for _r7_
Reported in version _r7_pre4_ on platform _i386_
Assigned to: David Korth

On 2009-10-08 15:06:14 -0400, David Korth wrote:

> On some systems, Gens/GS's ASPI handler causes an illegal operation on startup. I managed to "fix" it in a VM by allocating more memory than necessary, but this seems to be reappearing on other systems.
> 
> Potential workaround: Write a temporary file before starting ASPI, and delete it if initialization was successful. If the program crashes, then temporary file will exist when Gens/GS is started again, in which case ASPI will be disabled and a configuration setting will be saved, which disables ASPI.
> 
> Also, try doing some debugging to figure out what exactly is crashing. (Invalid memory access? gcc bug?)
> 
> The tentative port to libcdio will eliminate all the ASPI code from Gens/GS, so that may help things.

On 2009-10-11 12:58:14 -0400, David Korth wrote:

> **\* Bug 9 has been marked as a duplicate of this bug. ***

On 2009-10-16 22:15:12 -0400, David Korth wrote:

> Hopefully, this will fix it! :)
> 
> commit SHA: f95f8a3284562e508907aaeba72ae612d377ed1d
> Author: David Korth gerbilsoft@verizon.net
> Date:   Fri Oct 16 22:10:59 2009 -0400
> 
> ```
> segacd/aspi.h: Increased the buffer size in SRB_GDEVBlock.
> 
> This seems to fix the problem with ASPI on Win98 SE with the original
> ASPI driver. I'm not exactly sure what is going on here. (Adaptec ASPI
> v4.60 doesn't seem to have this issue.)
> 
> Also, cleaned up some stuff in segacd/cd_aspi_win32.cpp.
> ```
